plugins/woocommerce/src/StoreApi/Routes/V1/AbstractCartRoute.php

```diff
@@ -111,7 +111,13 @@ protected function is_update_request( \WP_REST_Request $request ) {
 	 * @return \WP_REST_Response
 	 */
 	public function get_response( \WP_REST_Request $request ) {
```

Comment:
As stated in slack, could this work instead? `updated_cart()` is too expensive not to be avoided if possible.
Needs some investigation to check if `updated_cart()` bubbles up any error we don't want in a response.

```php
public function get_response( \WP_REST_Request $request ) {

		try {
			$this->load_cart_session( $request );

			$nonce_check = $this->requires_nonce( $request ) ? $this->check_nonce( $request ) : null;

			if ( is_wp_error( $nonce_check ) ) {
				throw new RouteException(
					$nonce_check->get_error_message(),
					$nonce_check->get_error_code(),
					$nonce_check->get_error_data()['status'],
					$nonce_check->get_error_data()
				);
			}

			$response = $this->get_response_by_request_method( $request );

			// For update requests, this will recalculate cart totals and sync draft orders with the current cart.
			if ( $this->is_update_request( $request ) ) {
				$this->cart_updated( $request );
			}
		} catch ( RouteException $error ) {
			$response = $this->get_route_error_response( $error->getErrorCode(), $error->getMessage(), $error->getCode(), $error->getAdditionalData() );
		} catch ( \Exception $error ) {
			$response = $this->get_route_error_response( 'woocommerce_rest_unknown_server_error', $error->getMessage(), 500 );
		}

		return $this->add_response_headers( rest_ensure_response( $response ) );
	}
```

Reply:
update, but we always need to convert response to proper error from its wp_error